src/html/listener/mod.rs

```diff
@@ -0,0 +1,168 @@
+            .map(|input: TextAreaElement| input.value())
+            .ok(),
+    );
+    #[cfg(feature = "web_sys")]
+    let (v1, v2) = (
+        this.dyn_ref().map(|input: &InputElement| input.value()),
+        this.dyn_ref().map(|input: &TextAreaElement| input.value()),
+    );
+    let v3 = this.text_content();
+    let value = v1.or(v2).or(v3)
+        .expect("only an InputElement or TextAreaElement or an element with contenteditable=true can have an oninput event listener");
+    InputData { value }
+}
+
+fn onchange_handler(this: &Element) -> ChangeData {
+    match this.node_name().as_ref() {
+        "INPUT" => {
+            #[cfg(feature = "std_web")]
+            let input: InputElement = this.clone().try_into().unwrap();
+            #[cfg(feature = "web_sys")]
+            let input: &InputElement = this.dyn_ref().unwrap();
+            let is_file = input
+                .get_attribute("type")
+                .map(|value| value.eq_ignore_ascii_case("file"))
+                .unwrap_or(false);
+            if is_file {
+                #[cfg(feature = "std_web")]
+                let files: FileList = js!( return @{input}.files; ).try_into().unwrap();
+                #[cfg(feature = "web_sys")]
+                let files: FileList = input.files().unwrap();
+                ChangeData::Files(files)
+            } else {
+                #[cfg(feature = "std_web")]
+                return ChangeData::Value(input.raw_value());
+                #[cfg(feature = "web_sys")]
+                ChangeData::Value(input.value())
+            }
+        }
+        "TEXTAREA" => {
+            #[cfg(feature = "std_web")]
+            let tae: TextAreaElement = this.clone().try_into().unwrap();
+            #[cfg(feature = "web_sys")]
+            let tae: &TextAreaElement = this.dyn_ref().unwrap();
+            ChangeData::Value(tae.value())
+        }
+        "SELECT" => {
+            #[cfg(feature = "std_web")]
+            let se: SelectElement = this.clone().try_into().unwrap();
+            #[cfg(feature = "web_sys")]
+            let se = this.dyn_ref::<SelectElement>().unwrap().clone();
+            ChangeData::Select(se)
+        }
+        _ => {
+            panic!("only an InputElement, TextAreaElement or SelectElement can have an onchange event listener");
+        }
+    }
+}
+
+/// Handler to an event listener, only use is to cancel the event.
+// We can't use `gloo`s implementation because it cancels the event upon dropping the handler, but
+// we want the event to only be cancelled when the user desires. The main issue here is that
+// `wasm-bindgen` doesn't support moving a closure to WASM, so the closure has to be "forgotten"
+// and not be dropped, therefore the use of `ManuallyDrop` here.
+#[cfg(feature = "web_sys")]
+#[derive(Debug)]
+pub struct EventListenerHandle {
+    pub(crate) target: EventTarget,
+    pub(crate) r#type: &'static str,
+    pub(crate) callback: ManuallyDrop<Closure<dyn Fn(Event)>>,
```

Comment:
If you just wrap a `gloo::EventListener` in `ManuallyDrop` then the listener won't be removed on drop.

Reply:
Yes, thats what I did **not** want to do, because leaking a `Closure` is better then leaking a whole `EventListener`.
But thats solved now that I know that `EventListener` implements its own `forget`, thanks for that.

Reply:
If it doesn't do what you want please open an issue in `gloo`.

Reply:
Done.